src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -529,6 +530,46 @@ public long size() throws IOException {
         }
     }
 
+    public int available() throws IOException {
+        ensureOpen();
+        synchronized (positionLock) {
+            int a = -1;
+            int ti = -1;
+            try {
+                beginBlocking();
+                ti = threads.add();
+                if (!isOpen())
+                    return -1;
+                a = nd.available(fd);
+            } finally {
+                threads.remove(ti);
+                endBlocking(a > -1);
+            }
+            return a;
+        }
+    }
+
+    public boolean isOther() throws IOException {
+        ensureOpen();
+        int ti = -1;
+        Boolean isOther = null;
+        try {
+            beginBlocking();
+            ti = threads.add();
+            if (!isOpen())
+                return false;
+            boolean attempted = Blocker.begin();
+            try {
+                return isOther = nd.isOther(fd);
+            } finally {
+                Blocker.end(attempted);
```

Comment:
No need for Blocker begin/end here, we only use for direct I/O or file locking.

Reply:
Fixed in a163d8d.